extensions/2.0/Khronos/KHR_implicit_shapes/README.md

```diff
@@ -0,0 +1,147 @@
+# KHR_implicit_shapes
+
+## Contributors
+
+* Eoin Mcloughlin, Microsoft, <mailto:eomcl@microsoft.com>
+* Rory Mullane, Microsoft, <mailto:romul@microsoft.com>
+* George Tian, Microsoft, <mailto:geotian@microsoft.com>
+* Aaron Franke, Godot Engine, <mailto:arnfranke@yahoo.com>
+* Eric Griffith, Meta, <mailto:ericgriffith@meta.com>
+* Janine Liu, Cesium, <mailto:janine@cesium.com>
+* Sean Lilley, Cesium, <mailto:sean@cesium.com>
+
+## Status
+
+Release Candidate
+
+## Dependencies
+
+Written against glTF 2.0 spec.
+
+## Overview
+
+This extension adds data definitions for implicit shapes. This extension does not mandate any particular behaviour for these objects aside from a description of their geometry. These types are to be used in combination with other extensions which should specify the behavior of these types.
+
+## glTF Schema Updates
+
+### Shapes
+
+This extension provides a set of document-level objects, which can be referenced by objects in a glTF document.
+
+Shapes are defined within a dictionary property in the glTF scene description file, by adding an `extensions` property to the top-level glTF 2.0 object and defining a `KHR_implicit_shapes` property with a `shapes` array inside it.
+
+Each shape defines a mandatory `type` property which designates the type of shape, as well as an additional structure which provides parameterizations specific to that type. The following example defines a sphere.
+
+```javascript
+"extensions": {
+    "KHR_implicit_shapes" : {
+        "shapes": [
+            {
+                "sphere": { "radius": 0.1 },
+                "type": "sphere"
+            }
+        ]
+    }
+}
+```
+
+To describe the geometry which represents the object, shapes must define at most one of the following properties:
```

Comment:
Consider requiring the corresponding object (so it's always available in the Object Model).

Reply:
Do you mean that even if a shape was utilizing the default values, you'd want to require an empty object? e.g.

```
"shapes" = [
{
     "type": "sphere",
     "sphere": {}
}
...
```

One complication on REQUIREing one of the objects that other extensions may add additional types and may not wish to have a fallback:

```
"shapes" = [
{
     "type": "teapot",
     "extensions": {
            "ext_teapot_shape" : {
                       ....
            }
    }
}
...
```

For those types, we wouldn't want to require an unrelated object in the shape.

Reply:
Even when an object is syntactically empty it's not semantically empty because most shape properties have default values so, e.g., `"sphere": {}` actually means `"sphere": { "radius": 0.5 }`.

The extension should require an object matching the type so future extensions wouldn't require unrelated objects.